geminiScripts/pdfParser.py

```diff
@@ -0,0 +1,186 @@
+import requests
+import io
+import pdfplumber
+from docx import Document
+import pypandoc
+import os
+import json
+from structureGemini import process_and_structure_document
+import time
+from google.api_core.exceptions import ResourceExhausted
+from google.api_core.exceptions import DeadlineExceeded
+from trackScheme import (
+    calculate_hash,
+    load_previous_state,
+    save_current_state,
+    identify_changes,
+)
+
+
+
+base_file_path = os.path.join(os.path.dirname(__file__),'..','goaScraper','goa_pdf_link.json')
+absolute_file_path = os.path.abspath(base_file_path)
+output_file = os.path.join(os.path.dirname(__file__), 'structured_results.json')
+state_file = 'scheme_state.json'
+
+api_keys = [
+    "AIzaSyCwZDOJVmSi7hxrw-1oHadN__3-Qnh3-uE",
+    "AIzaSyDun0IvxbLRGtj484dh2H59ZJQY125pozc",
+    "AIzaSyDryM70GXyoxoZOUuT-oiJSEypjNohBGDI",
+    "AIzaSyAUdZiWi-TBgR5NK43_zs3G4yQEiMm9zpw",
+    "AIzaSyCtyRHNl2W0jsFA-yiA2jGaNkG6_zr2mzc",
+    "AIzaSyB4XEu-v61Jlxz_otvFPrQfJvPxCgm_FFs",
+    "AIzaSyDdahnDepkau1tx5gf4gFiDq0R_hxbwfcg"
+]
```

Comment:
@aryan7081 Don't hardcode API Keys in the repo, define Env variables instead and use them here.